### PR TITLE
Stored Procedures - call with unnamed args (parsing + PIG definition)

### PR DESCRIPTION
**Pull request by alancai98**
Adds parsing and PIG definition of stored procedure calls with the unnamed argument syntax defined in PartiQL spec issue #[17](https://github.com/partiql/partiql-spec/issues/17).

A valid call to a stored procedure as part of this PR is defined by `EXEC <symbol> [<expr>.*]` and is only a valid PartiQL statement when in the top level query.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lang/src/org/partiql/lang/ast/passes/AstWalker.kt [hunk 1]
```diff
@@ -116,6 +117,7 @@ open class AstWalker(private val visitor: AstVisitor) {
                     }
                 }
                 is CreateTable, is DropTable, is DropIndex -> case { }
+                is Exec         -> throw UnsupportedOperationException("EXEC clause is not supported by the V0 AST")
```
Comment (dlurton) [inline lang/src/org/partiql/lang/ast/passes/AstWalker.kt:120]: 1.  Copy-pasta from the `V0` serializer?
1. We do need to support this in the `AstWalker` for time being, so we shouldn't be throwing an exception here.  Probably, an empty `case { }` like the previous line is appropriate.

File: lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt [hunk 2]
```diff
@@ -278,6 +278,8 @@ internal class EvaluatingCompiler(
             is CreateIndex,
             is DropIndex,
             is DropTable -> compileDdl(expr)
+            is Exec              -> compileExec(expr)
```
Comment (dlurton) [inline lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt:281]: Nit: formatting.

**Review by dlurton — CHANGES_REQUESTED** [review]

File: lang/src/org/partiql/lang/ast/passes/AstWalker.kt (hunk 1 above)
Comment (alancai98) [reply lang/src/org/partiql/lang/ast/passes/AstWalker.kt]: 1. Yeah that was copied from the `AstSerialization` code.
2. Changed to `case { }`.

File: lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt (hunk 2 above)
Comment (alancai98) [reply lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt]: Fixed. Thanks

**Comment by alancai98** [comment]
New revision fixes previous feedback. Also changes `func_name`/`funcName` usages to `procedure_name`/`procedureName`. And adds check that the args aren't surrounded in parens (i.e. calling `EXEC sproc_foo(arg0, arg1)` now throws an error).

File: lang/src/org/partiql/lang/ast/passes/AstWalker.kt [hunk 3]
```diff
@@ -16,6 +16,7 @@ package org.partiql.lang.ast.passes
 
 import org.partiql.lang.ast.*
 import org.partiql.lang.util.*
+import java.lang.UnsupportedOperationException
```
Comment (therapon) [inline lang/src/org/partiql/lang/ast/passes/AstWalker.kt:19]: is this import needed? looks like a left-over

Comment (alancai98) [reply lang/src/org/partiql/lang/ast/passes/AstWalker.kt]: Yes, that is leftover. Removed in the new commit. Thanks!

**Review by dlurton — APPROVED** [review]

**Pull request merged**